### PR TITLE
use_data() improvements

### PR DESCRIPTION
**Pull request by krlmlr**
- Helper functions
- Unified code for data and sysdata
- Warn if duplicated objects
- Stop if no objects to save

### CODE REVIEW

File: R/infrastructure.R [hunk 1]
```diff
@@ -399,40 +399,75 @@ use_data <- function(..., pkg = ".", internal = FALSE, overwrite = FALSE,
   pkg <- as.package(pkg)
 
   to_save <- dots(...)
```
Comment (krlmlr) [inline R/infrastructure.R]: @hadley: Do you mind if I import `lazyeval` and implement a corresponding `use_data_` function? The above seems to be the only remaining instance of `dots()`.

Comment (hadley) [inline R/infrastructure.R]: Doesn't seem worth it to me - `use_data()` is purely for interactive use.

File: R/infrastructure.R [hunk 2]
```diff
@@ -399,40 +399,75 @@ use_data <- function(..., pkg = ".", internal = FALSE, overwrite = FALSE,
...
+    stop("Nothing to save", call. = FALSE)
+  }
+
   is_name <- vapply(to_save, is.symbol, logical(1))
   if (any(!is_name)) {
     stop("Can only save existing named objects", call. = FALSE)
   }
+}
+
+get_objs_from_to_save <- function(to_save) {
```
Comment (hadley) [inline R/infrastructure.R]: How about integrating this with `check_to_save()`?

File: R/infrastructure.R [hunk 3]
```diff
@@ -399,40 +399,75 @@ use_data <- function(..., pkg = ".", internal = FALSE, overwrite = FALSE,
...
+  if (length(duplicated_objs) > 0L) {
+    objs <- unique(objs)
+    warning("Saving duplicates only once: ",
+            paste(names(duplicated_objs), collapse = ", "),
+            call. = FALSE)
+  }
+  objs
+}
 
+get_save_info_for_objs <- function(pkg, internal, objs) {
```
Comment (hadley) [inline R/infrastructure.R]: I'd rather just inline this inside `use_data()`

**Comment by krlmlr** [comment]
Inlined functions, merged master, added NEWS. I'd rather not rebase, because I have other internal branches based on this.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**